internal/worker/worker.go

```diff
@@ -0,0 +1,54 @@
+package worker
+
+import (
+	"context"
+	"sync"
+)
+
+type Task interface {
+	Process()
+}
+
+type WorkerPool struct {
+	taskChan    chan Task
+	wg          sync.WaitGroup
```

Comment:
It is always better to have a pointer to a mutext or to a wait group, as it gives a better guarantee that the structure in question is not copied after initialization